README.md
# Thiago Villa's website

#todo short desc

## Install

You need [Node v10+](https://nodejs.org/en/download/) and [Gatsby](https://www.gatsbyjs.com/get-started/).

```shell
git clone https://github.com/thiagovilla/thiagovilla.github.io.git
cd thiagovilla.github.io
npm install
npm run develop
```

## Goal

#todo

## Skills

#todo

## Stack

- Front-end framework: [GatsbyJS](https://gatsbyjs.com)
- UI framework: [React](https://reactjs.org) (part of Gatsby itself)
- Styling system: [Emotion](https://emotion.sh)
- Design system: [Tachyons](https://tachyons.io), if it "works" with Emotion
- Back-end (CMS): [DatoCMS](https://datocms.com)
- Deploy plataform: [GitHub Pages](https://pages.github.com)

## Features

#todo

## To do

- Design and code home page
- Design and code post page

## Changelog

See [CHANGELOG.md](CHANGELOG.md).

## Roadmap

- 🛠️ Connect DatoCMS
- 🛠️ Add Disqus comments
- 📅 (2021 Q1) Twitter feed on footer
- 📅 (2021 Q1) LinkedIn feed on footer
- 🤔 Add book, video, online course summaries
- 💡 Combine Twitter, LinkedIn, and GitHub feeds
- 💡 GitHub activity feed on footer
- 💡 Add "Was this helpful?" to posts
- 💡 Library of curated UI patterns (epic 🧙‍♂️)

Legend: 🛠️ working on it, 📅 scheduled, 🤔 considering, 💡 idea

## Creative process

See [CREATIVE_PROCESS.md](CREATIVE_PROCESS.md).

## About me

#todo

#todo add legal, credits, acknowledgments


CHANGELOG.md
# Changelog

## Version `0.1.1`

Installed and set up DatoCMS source plugin with my DatoCMS read-only API key. Added sample GraphQL query to `index.js` for last five posts. Also prettified `index.js`.

## Version `0.1.0`

Created a blank Gatsby app (no plugins). Replaced `index.js`'s contents with a "coming soon" placeholder message–linked to [my LinkedIn profile](https://linkedin.com/in/thiagovilla89), at least 😎. Then installed the GitHub pages and Google Analytics plugins. Added GitHub's `CNAME` file with my domain `thiagovilla.com` to the `/static` folder. Also set up the GA plugin with my tracking ID.

## Before version `0.1.0`

#todo


webpack-runtime-f5254048e5526a57a3da.js
!function(e){function r(r){for(var n,a,i=r[0],c=r[1],l=r[2],p=0,s=[];p<i.length;p++)a=i[p],Object.prototype.hasOwnProperty.call(o,a)&&o[a]&&s.push(o[a][0]),o[a]=0;for(n in c)Object.prototype.hasOwnProperty.call(c,n)&&(e[n]=c[n]);for(f&&f(r);s.length;)s.shift()();return u.push.apply(u,l||[]),t()}function t(){for(var e,r=0;r<u.length;r++){for(var t=u[r],n=!0,i=1;i<t.length;i++){var c=t[i];0!==o[c]&&(n=!1)}n&&(u.splice(r--,1),e=a(a.s=t[0]))}return e}var n={},o={0:0},u=[];function a(r){if(n[r])return n[r].exports;var t=n[r]={i:r,l:!1,exports:{}};return e[r].call(t.exports,t,t.exports,a),t.l=!0,t.exports}a.e=function(e){var r=[],t=o[e];if(0!==t)if(t)r.push(t[2]);else{var n=new Promise((function(r,n){t=o[e]=[r,n]}));r.push(t[2]=n);var u,i=document.createElement("script");i.charset="utf-8",i.timeout=120,a.nc&&i.setAttribute("nonce",a.nc),i.src=function(e){return a.p+""+({2:"component---src-pages-404-js",3:"component---src-pages-index-js"}[e]||e)+"-"+{2:"919de4b44033d32068dd",3:"6843281d23f4ccaf8536"}[e]+".js"}(e);var c=new Error;u=function(r){i.onerror=i.onload=null,clearTimeout(l);var t=o[e];if(0!==t){if(t){var n=r&&("load"===r.type?"missing":r.type),u=r&&r.target&&r.target.src;c.message="Loading chunk "+e+" failed.\n("+n+": "+u+")",c.name="ChunkLoadError",c.type=n,c.request=u,t[1](c)}o[e]=void 0}};var l=setTimeout((function(){u({type:"timeout",target:i})}),12e4);i.onerror=i.onload=u,document.head.appendChild(i)}return Promise.all(r)},a.m=e,a.c=n,a.d=function(e,r,t){a.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},a.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},a.t=function(e,r){if(1&r&&(e=a(e)),8&r)return e;if(4&r&&"object"==typeof e&&e&&e.__esModule)return e;var t=Object.create(null);if(a.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:e}),2&r&&"string"!=typeof e)for(var n in e)a.d(t,n,function(r){return e[r]}.bind(null,n));return t},a.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return a.d(r,"a",r),r},a.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},a.p="/",a.oe=function(e){throw console.error(e),e};var i=window.webpackJsonp=window.webpackJsonp||[],c=i.push.bind(i);i.push=r,i=i.slice();for(var l=0;l<i.length;l++)r(i[l]);var f=c;t()}([]);
//# sourceMappingURL=webpack-runtime-f5254048e5526a57a3da.js.map

component---src-pages-index-js-6843281d23f4ccaf8536.js
(window.webpackJsonp=window.webpackJsonp||[]).push([[3],{RXBc:function(e,t,n){"use strict";n.r(t);var i=n("q1tI"),l={color:"#232129",padding:"96px",fontFamily:"-apple-system, Roboto, sans-serif, serif"},a={marginTop:0,marginBottom:64,maxWidth:320},o={marginBottom:48},m={marginBottom:96,paddingLeft:0},s={color:"#8954A8",fontWeight:"bold",fontSize:"16px",verticalAlign:"5%"},r=Object.assign({},s,{listStyleType:"none",marginBottom:24});t.default=function(){return i.createElement("main",{style:l},i.createElement("title",null,"Thiago Villa"),i.createElement("h1",{style:a},"Thiago Villa"),i.createElement("p",{style:o},"Coming soon!"," ",i.createElement("span",{role:"img","aria-label":"Sunglasses smiley emoji"},"😎")),i.createElement("ul",{style:m},i.createElement("li",{style:r},"Find me on ",i.createElement("a",{style:s,href:"https://linkedin.com/in/thiagovilla89"},"LinkedIn"))))}}}]);
//# sourceMappingURL=component---src-pages-index-js-6843281d23f4ccaf8536.js.map

src/pages/index.js
import * as React from "react";
import { graphql } from "gatsby";

// styles
const pageStyles = {
  color: "#232129",
  padding: "96px",
  fontFamily: "-apple-system, Roboto, sans-serif, serif",
};
const headingStyles = {
  marginTop: 0,
  marginBottom: 64,
  maxWidth: 320,
};
const paragraphStyles = {
  marginBottom: 48,
};
const listStyles = {
  marginBottom: 96,
  paddingLeft: 0,
};
const linkStyle = {
  color: "#8954A8",
  fontWeight: "bold",
  fontSize: "16px",
  verticalAlign: "5%",
};
const docLinkStyle = {
  ...linkStyle,
  listStyleType: "none",
  marginBottom: 24,
};

// markup
const IndexPage = ({ data }) => {
  return (
    <main style={pageStyles}>
      <title>Thiago Villa</title>
      <h1 style={headingStyles}>Thiago Villa</h1>
      <p style={paragraphStyles}>
        Coming soon!{" "}
        <span role="img" aria-label="Sunglasses smiley emoji">
          😎
        </span>
      </p>
      <ul style={listStyles}>
        <li style={docLinkStyle}>
          Find me on{" "}
          <a style={linkStyle} href="https://linkedin.com/in/thiagovilla89">
            LinkedIn
          </a>
        </li>
        {data.allDatoCmsPost.edges.map(({ node: post }) => (
          <li>
            {post.title} {post.meta.publishedAt}
          </li>
        ))}
      </ul>
    </main>
  );
};

export default IndexPage;

export const query = graphql`
  query Last5Posts {
    allDatoCmsPost(
      sort: { fields: [meta___publishedAt], order: DESC }
      limit: 5
    ) {
      edges {
        node {
          meta {
            publishedAt
          }
          title
        }
      }
    }
  }
`;
